### ISSUE TITLE
if no argument `os._exit()` comes after `return`, pyright doesn't notice us

### ISSUE THREAD

**Issue by amirsoroush** [microsoft/pyright#10260]
**Describe the bug**

`os._exit()` needs 1 argument. Pyright correctly catches the missing argument when `os._exit()` is left alone, but when it comes after `return` it ignores that.
Yes I agree that this is a low priority issue and most python programmers don't even notice it but it may lead to undesired behavior. (see the code)

**Code or Screenshots**

```python
import os

def foo():
    try:
        os._exit()
    finally:
        return False

foo()
print("this is running!")
```
the output of running `pyright`:

```none
... :5:9 - error: Argument missing for parameter "status" (reportCallIssue)
1 error, 0 warnings, 0 informations
```

Now look at this:

```python
import os

def foo():
    try:
        return os._exit()
    finally:
        return False

foo()
print("this is running!")
```
the output of `pyright`:

```none
0 errors, 0 warnings, 0 informations
```

The problem is that if you run the second code, you'll see the `"this is running!"` output. The reason is that `os._exit()` raised a `TypeError` (because it needed an argument) so the finally block executed and the exception is swallowed by `return False`.

**Comment by erictraut** [comment]
Thanks for reporting. I agree this is a bug.

Here's an even simpler code snippet that repros the problem.

```python
from typing import NoReturn

def func(x: int) -> NoReturn: ...

def test():
    return func()
```
